File: eniato/apps/accounts/factories/financial_institution_factory.py
from apps.accounts.domain import FinancialInstitution


class FinancialInstitutionFactory:

    def create_from_data(self, data):
        attributes = {}
        return FinancialInstitution(attributes)

    def create_from_model(self, model):
        return FinancialInstitution(
            {
                'id': model.id,
                'code': model.code,
                'name': model.name,
                'logo': model.logo.url,
            }
        )

File: eniato/vue-app/src/apps/account/factories/FinancialInstitutionFactory.js
import { FinancialInstitution } from '../domain/FinancialInstitution'

export class FinancialInstitutionFactory {
  buildFromDTO (dto) {
    return new FinancialInstitution({
      id: dto.id,
      name: dto.name,
      code: dto.code,
      logo: dto.logo
    })
  }
}
